packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -1059,6 +1069,38 @@ contract VotingV2 is
         return spamDeletionProposals[spamDeletionRequestId];
     }
 
+    function signalOnEmergencyAction(
+        bytes32 identifier,
+        uint256 actionTime,
+        bytes memory ancillaryData
+    ) public {
+        EmergencyAction storage action = emergencyActions[_encodePriceRequest(identifier, actionTime, ancillaryData)];
+
+        if (!action.accountSignaled[msg.sender]) {
+            action.accountSignaled[msg.sender] = true;
+            action.cumulativeSignaled += getVoterStake(msg.sender);
+        }
```

Comment:
Could someone multiply their voting power by doing the following?

1. I have 25% of the cumulative stake.
2. I stake from address 0x1234 and call signalOnEmergencyAction().
3. I unstake, which might take a week.
4. After a week, I send all my tokens to 0x5678 and stake with that address.
5. I call signalEmergencyAction() from 0x5678.

Even though I only had 25% of the cumulative stake, I was able to amplify my voting power to 50% of the cumulative stake.

Does that attack make sense? Do you agree that something like that is possible?

Reply:
I think this is a very good observation, In my opinion this is feasible and we should prevent it.

Reply:
hhmm. so the assumption with this is that the staking/unstaking logic is broken but you right that you could do something like what you describe to get through normal price actions.

to prevent this we can prevent someone from staking/unstaking if they've voted on an unexecuted emergency tx.

Reply:
added! PTAL